src/lib/web-component/filter-timestamp/filter-timestamp-element.js

```diff
@@ -58,29 +62,64 @@ export class FilterTimestamp extends HTMLElement {
     );
   }
 
-  onClearFilter = () => {
-    this.startTimeLabel.textContent = 'From Beginning';
-    this.stopTimeLabel.textContent = 'Up to End';
-    this.filterContainer.hidden = true;
+  onClearFilter(clearStart = true, clearStop = true) {
+    if (clearStart) {
+      this.startTimeLabel.textContent = 'From Beginning';
+      this.clearStartTimeBtn.hidden = true;
+      delete this.timeStamp.start;
+    }
+
+    if (clearStop) {
+      this.stopTimeLabel.textContent = 'Up to End';
+      this.clearStopTimeBtn.hidden = true;
+      delete this.timeStamp.stop;
+    }
+
+    if (Object.keys(this.timeStamp).length === 0) {
+      this.timeStamp = null;
+    }
+
+    if (!this.timeStamp) {
+      this.filterContainer.hidden = true;
+    }
 
-    this.timeStamp = null;
     this.dispatchFilterChange();
-  };
+  }
 
   onHiddenListener = () => {
     browser.menus.removeAll();
   };
 
+  handleEvent(event) {
+    if (event.type === 'click') {
+      switch (event.target) {
+        case this.clearFilterBtn:
+          this.onClearFilter();
+          break;
+        case this.clearStartTimeBtn:
+          this.onClearFilter(true, false);
+          break;
+        case this.clearStopTimeBtn:
+          this.onClearFilter(false, true);
+          break;
+      }
+    }
+  }
+
   connectedCallback() {
     browser.menus.onClicked.addListener(this.setFilterRange);
     browser.menus.onHidden.addListener(this.onHiddenListener);
-    this.clearBtn.addEventListener('click', this.onClearFilter);
+    this.clearFilterBtn.addEventListener('click', this);
+    this.clearStartTimeBtn.addEventListener('click', this);
+    this.clearStopTimeBtn.addEventListener('click', this);
```

Comment:
you should be able to subscribe a single click event listener (e.g. on `this.filterContainer`) without any additional changes (the `handleEvent` is already conditioned on the `event.target` property).

Reply:
done :-) Thanks.